SpatialGDK/Source/SpatialGDK/Private/Interop/Startup/DefaultServerWorkerStartupHandler.h

```diff
@@ -1,95 +1,45 @@
 ﻿#pragma once
 
-#include "EngineClasses/SpatialVirtualWorkerTranslator.h"
-#include "Interop/ClaimPartitionHandler.h"
-#include "Interop/Connection/SpatialWorkerConnection.h"
-#include "Interop/CreateEntityHandler.h"
-#include "Interop/SkeletonEntityCreationStep.h"
-#include "SpatialCommonTypes.h"
+#include "Containers/UnrealString.h"
+#include "Templates/SharedPointer.h"
+
+#include "Interop/Startup/SpatialStartupCommon.h"
 
 class USpatialNetDriver;
 class UGlobalStateManager;
 
 namespace SpatialGDK
 {
-class ViewCoordinator;
+class ISpatialOSWorker;
+
+struct FInternalState;
+
+struct FInitialSetup
+{
+	int32 ExpectedServerWorkersCount;
+};
 
-class FSpatialServerStartupHandler
+class FSpatialServerStartupHandler final
 {
 public:
-	struct FInitialSetup
-	{
-		int32 ExpectedServerWorkersCount;
-	};
 	explicit FSpatialServerStartupHandler(USpatialNetDriver& InNetDriver, const FInitialSetup& InSetup);
+	~FSpatialServerStartupHandler();
 	bool TryFinishStartup();
 	FString GetStartupStateDescription() const;
 
 private:
-	void SpawnPartitionEntity(Worker_EntityId PartitionEntityId, VirtualWorkerId VirtualWorkerId);
-	bool TryClaimingStartupPartition();
-
-	bool bCalledCreateEntity = false;
-	TOptional<ServerWorkerEntityCreator> WorkerEntityCreator;
-	TOptional<Worker_EntityId> WorkerEntityId;
-
-	TArray<Worker_EntityId_Key> WorkerEntityIds;
-
-	bool bHasGSMAuth = false;
-
-	bool bIsRecoveringOrSnapshot = false;
-
-	bool bStartedWorkerPartitionQuery = false;
-	FEntityQueryHandler PartitionQueryHandler;
-
-	bool bHasCalledPartitionEntityCreate = false;
-	FCreateEntityHandler EntityHandler;
-	TArray<Worker_PartitionId> WorkerPartitions;
-
-	TSet<Worker_PartitionId> PartitionsToCreate;
-
-	TOptional<VirtualWorkerId> LocalVirtualWorkerId;
-	TOptional<Worker_PartitionId> LocalPartitionId;
-
-	FClaimPartitionHandler ClaimHandler;
-
-	TOptional<FSkeletonEntityCreationStartupStep> SkeletonEntityStep;
-
-	ViewCoordinator& GetCoordinator();
-	const ViewCoordinator& GetCoordinator() const;
-	const TArray<Worker_Op>& GetOps() const;
+	ISpatialOSWorker& GetCoordinator();
```

Comment:
Nit that can be ignored as it's a driveby. More of a thought for later, we should probably call these GetWorkerInterface or something unless actually returning the coordinator itself.